R/Yisheng-L2999.ccs_shrwt_CHINA.R

```diff
@@ -0,0 +1,77 @@
+#' module_gcam.china_L2999.ccs_shrwt_CHINA
+#'
+#' Assign shareweight assumptions of CCS to subtechnologies by China province
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L9999.StubTechInterpOverwrite_CHINA}, \code{L9999.StubTechShrwt_CHINA}. The corresponding file in the
+#' original data system was \code{L2999.ccs_shrwt_CHINA.R} (gcam-china level2).
+#' @details Assign shareweight assumptions of CCS to subtechnologies by China province. Remove N fertilizer sector from Tibet(XZ) province.
+#' @importFrom dplyr filter
+#' @author YishengSun July 2019
+module_gcam.china_L2999.ccs_shrwt_CHINA <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c(FILE = "gcam-china/province_names_mappings",
+             FILE = "gcam-china/A999.StubTechShrwt_ccs",
+             FILE = "gcam-china/A999.StubTechInterpOverwrite"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L9999.StubTechShrwt_CHINA",
+             "L9999.StubTechInterpOverwrite_CHINA"))
+  } else if(command == driver.MAKE) {
+
+
+    all_data <- list(...)[[1]]
+
+    # Silence package checks
+    region <- supplysector <- subsector <- subsector.name <- stub.technology <- year <-
+      share.weight <- delete <- apply.to <- from.year <- to.year <- interpolation.function <- NULL
+
+    # Load required inputs
+    province_names_mappings <- get_data(all_data, "gcam-china/province_names_mappings")
+    StubTechShrwt_ccs <- get_data(all_data, "gcam-china/A999.StubTechShrwt_ccs")
+    StubTechInterpOverwrite <- get_data(all_data, "gcam-china/A999.StubTechInterpOverwrite")
+
+    # ===================================================
+    # Apply new ccs shareweights to all provinces
+    StubTechShrwt_ccs_1 <- write_to_all_provinces(StubTechShrwt_ccs, c("region", "supplysector", "subsector.name", "stub.technology", "year", "share.weight"))
```

Comment:
Note: after merging in this, will need to add provinces parameter in, since we updated the function.